CONTRIBUTING.md

```diff
@@ -14,6 +14,54 @@ Thanks for taking the plunge!
   (`Pkg.add(name="DataFrames", rev="main")`) is a good gut check and can streamline the process,
   along with including the first two lines of output from `versioninfo()`
 
+## Modifying an existing docstring in `src/`
+
+All docstrings are written inline above the methods or types they are associated with and can be found by clicking on the `source` link that appears below each docstring in the HTML file. The steps needed to make a change to an existing docstring are listed below:
+
+* Find the docstring in `src/`;
+* Update the text in the docstring;
+* run `make docs` from the root directory;
+* check the output in `doc/_build/html/` to make sure the changes are correct;
+* commit your changes and open a pull request.
+
+## Adding a new docstring to `src/`
+
+The steps required to add a new docstring are listed below:
+* find a suitable definition in `src/` that thedocstring will be most applicable to;
+* add a doctring above the definition;
+* find a suitable `@docs` code block in one of the `docs/src/` files where you would like the docstring to appear;
+* add the name of the definition to the `@docs` code block. For example, with a docstring added to a function `bar`
+```
+"..."
+function bar(args...)
+    # ...
+end
+```
+
+you would add the name `bar` to a `@docs` block in `docs/src/`
+````
+```@docs
+foo
+bar # <-- Added this one.
+baz
+```
+````
+
+* run `make docs` from the root directory;
+* check the output in `docs/_build/html` to make sure the changes are correct;
+* commit your changes and open a pull request.
+
+## Doctests
+
+Examples written within docstrings can be used as testcases known as `"doctests"` by annotating code blocks with jldoctest.
+````
+```jldoctest
+julia> uppercase("Docstring test")
+"DOCSTRING TEST"
+```
+````
+A doctest needs to match an interactive REPL including the julia> prompt. To run doctests you need to run make -C doc doctest=true from the root directory. It is recommended to add the header # Examples above the doctests.
```

Comment:
```suggestion
A doctest needs to match an interactive REPL including the `julia>` prompt. To run doctests you need to run `make -C doc doctest=true` from the root directory.

It is recommended to add the header `# Examples` above the doctests.
```

Reply:
is the `make -C doc doctest=true` ineede the command to run?

Reply:
If someone doesn't want to change `doctest=true` in make.jl

Reply:
but we have `doctest=true` in make.jl - right?